connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
@kkonstantine Yeah, that makes sense. I think there is definitely value in making connector startup/shutdown parallel at least. It probably makes sense for task shutdown also since we can't actually enforce that `stop()` is asynchronous. I'm wondering if we pushed startup/shutdown invocation into the `Worker` and used the same executor if task startup would become a little more natural (instead of executing a callable in order to submit a runnable, we'd just submit the runnable). What do you think?